client/ayon_resolve/plugins/create/create_editorial_package.py

```diff
@@ -0,0 +1,46 @@
+from pprint import pformat
+import json
+from ayon_core.pipeline.create.legacy_create import LegacyCreator
+
+from ayon_resolve.api import lib
+
+
+class CreateEditorialPackage(LegacyCreator):
+    """Create Editorial Package."""
+
+    name = "editorial_pkg"
+    label = "Editorial Package"
+    product_type = "editorial_pkg"
+    icon = "camera"
+    defaults = ["Main"]
+
+    def process(self):
+        """Process the creation of the editorial package."""
+        if (self.options or {}).get("useSelection"):
+            # if selected is on then get only timeline from media pool
+            # selection
+            self.convert_timeline_to_editorial_package_instance()
+        else:
+            # if selected is on then get only active timeline
+            self.convert_timeline_to_editorial_package_instance()
+
+    def convert_timeline_to_editorial_package_instance(self):
+        """Convert timeline to editorial package instance."""
+        current_timeline = lib.get_current_timeline()
+        timeline_name = current_timeline.GetName()
```

Comment:
This will error if there's no current timeline.
```suggestion
        current_timeline = lib.get_current_timeline()
        if not current_timeline:
            raise RuntimeError("Make sure to have an active current timeline.")
        timeline_name = current_timeline.GetName()
```